lib/mprintf.c

```diff
@@ -964,7 +964,7 @@ static int dprintf_formatf(
 #endif
         /* NOTE NOTE NOTE!! Not all sprintf implementations return number of
            output characters */
-        (sprintf)(work, formatbuf, p->data.dnum);
+        (snprintf)(work, formatbuf, p->data.dnum);
```

Comment:
1. This is not how `snprintf`  works
2. `snprintf` is not in C89 and thus not always available
3. This use of `sprintf()` is the only one in libcurl and has been reviewed and vetted

Reply:
okay, then we have to use `CFLAGS: "-Wno-deprecated-declarations"` for the m1 arm CI to build it.
Otherwise this error will be shown.

Reply:
Or we fix it by checking in configure if snprintf exists and make sure to use it conditionally and correctly in `mprintf.c`

Reply:
Let me give it a shot

Reply:
#9570

Reply:
IIRC, does not `snprintf` take the `work` size in characters as the second argument?

Reply:
Yes, hence my comment above:

> This is not how snprintf works

But I've already merged #9570 which had that corrected.